### PR TITLE
feat: allow to exec a process

### PR DESCRIPTION
**Pull request by caarlos0**
e.g.

```sh
sequin -- git status -sb
```

### CODE REVIEW

File: exec.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+// runInPty opens a new pty and runs the given command in it.
+// The returned file is the pty's file descriptor and must be closed by the
+// caller.
+func runInPty(c *exec.Cmd) (*os.File, error) {
+	//nolint: wrapcheck
+	return pty.StartWithAttrs(c, &pty.Winsize{
+		Cols: 80,
+		Rows: 10,
```
Comment (aymanbagabas) [inline exec.go:42]: We might want to get the size of the current tty and apply it here. We could also allow passing a custom size.

```suggestion
		Rows: 24,
```

Comment (aymanbagabas) [reply exec.go:42]: Oh, and this would only work in Unix. Probably want to use `x/xpty` instead

Comment (caarlos0) [reply exec.go]: should be better now! thanks for the review!

File: exec.go [hunk 2]
```diff
@@ -6,39 +6,40 @@ import (
...
 	var out bytes.Buffer
 	var errorOut bytes.Buffer
 	go func() {
 		_, _ = io.Copy(&out, pty)
 		errorOut.Write(out.Bytes())
 	}()
 
-	err = cmd.Wait()
-	if err != nil {
+	if err := cmd.Wait(); err != nil {
```
Comment (aymanbagabas) [inline exec.go:41]: We probably want to use [`xpty.WaitProcess`](https://github.com/charmbracelet/x/blob/main/xpty/xpty.go#L75) for Windows

Comment (aymanbagabas) [reply exec.go]: This should only be used on Windows so we need to put it in a `runtime.GOOS == "windows"` block

Comment (caarlos0) [reply exec.go]: ohh, is it worth it making it do a simple cmd.wait if not windows?

Comment (caarlos0) [reply exec.go]: done https://github.com/charmbracelet/x/pull/273

File: exec.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	"github.com/charmbracelet/x/term"
+	"github.com/charmbracelet/x/xpty"
+)
+
+func executeCommand(ctx context.Context, args []string) ([]byte, error) {
+	width, height, err := term.GetSize(os.Stdout.Fd())
+	if err != nil {
+		width = 80
+		height = 24
```
Comment (ccoVeille) [inline exec.go:18]: Please add constants such as

defaultTerminalWidth = 80
defaultTerminalHeight = 24

Or using "fallback terminal*"

**Review by ccoVeille — APPROVED** [review]

**Pull request merged**